core/modules/wsched/BlendScheduler.cc

```diff
@@ -153,19 +184,23 @@ bool BlendScheduler::ready() {
     return _ready();
 }
 
-/// Returns true when either scheduler has a command ready.
+/// Returns true when any sub-scheduler has a command ready.
 /// Precondition util::CommandQueue::_mx must be locked when this is called.
 bool BlendScheduler::_ready() {
-    auto groupReady = _group->ready();
-    auto scanReady = _scan->ready();
-    auto ready = groupReady || scanReady;
-    LOGS(_log, LOG_LVL_DEBUG, "BlendScheduler::_ready() groups("
-         << "r=" << groupReady
-         << ", q=" << _group->getSize()
-         << ", flight=" << _group->getInFlight() << ") "
-         << "scan(r=" << scanReady
-         << ", q=" << _scan->getSize()
-         << ", flight=" <<_scan->getInFlight() << ")");
+    std::ostringstream os;
+    bool ready = false;
+
+    // Get the total number of threads schedulers want reserved
+    int availableThreads = calcAvailableTheads();
+    for (auto sched : _schedulers) {
+        availableThreads = sched->applyAvailableThreads(availableThreads);
+        ready = sched->ready();
+        os << sched->getName() << "(r=" << ready << " sz=" << sched->getSize()
+           << " fl=" << sched-> getInFlight() << " avail=" << availableThreads << ") ";
```

Comment:
add `if (LOG_CHECK_LVL(_log, LOG_LVL_DEBUG))` around `os << ...` or maybe log it as individual message and don't collect it into one large string.